### PR TITLE
Handle out-of-memory failures in avifCodecCreate()

### PR DESCRIPTION
**Pull request by y-guyon**
See https://github.com/AOMediaCodec/libavif/issues/820.

### CODE REVIEW

File: src/read.c [hunk 1]
```diff
@@ -4061,8 +4061,7 @@ static avifResult avifCodecCreateInternal(avifCodecChoice choice, const avifTile
         choice = AVIF_CODEC_CHOICE_AVM;
     }
 #endif
-    AVIF_CHECKERR(tile->codecType == avifCodecTypeFromChoice(choice, AVIF_CODEC_FLAG_CAN_DECODE), AVIF_RESULT_NO_CODEC_AVAILABLE);
```
Comment (vrabaud) [inline src/read.c:4064]: why remove that check?

File: src/avif.c [hunk 2]
```diff
@@ -1087,13 +1087,14 @@ avifCodecChoice avifCodecChoiceFromName(const char * name)
     return AVIF_CODEC_CHOICE_AUTO;
 }
 
-avifCodec * avifCodecCreate(avifCodecChoice choice, avifCodecFlags requiredFlags)
+avifResult avifCodecCreate(avifCodecChoice choice, avifCodecFlags requiredFlags, avifCodec ** codec)
```
Comment (vrabaud) [inline src/avif.c:1090]: Is the idea to move APIs (like avifAlloc) to return avifResult? I could not agree more with the nodiscard I am adding.

**Review by vrabaud — APPROVED** [review]

File: src/read.c (hunk 1 above)
Comment (y-guyon) [reply src/read.c:4064]: Because it is part of `avifCodecCreate()` now.

File: src/avif.c (hunk 2 above)
Comment (y-guyon) [reply src/avif.c:1090]: The reason here is that `avifCodecCreate()` can fail in multiple ways (no codec available or out of memory). `avifResult` is also convenient for the calling site.

**Pull request merged**